Gemfile

```diff
@@ -3,9 +3,9 @@ source ENV['GEM_SOURCE'] || "https://rubygems.org"
 gemspec
 
 def location_for(place, fake_version = nil)
-  if place =~ /^(git[:@][^#]*)#(.*)/
+  if place.to_s =~ /^(git[:@][^#]*)#(.*)/
     [fake_version, { git: $1, branch: $2, require: false }].compact
-  elsif place =~ /^file:\/\/(.*)/
+  elsif place.to_s =~ /^file:\/\/(.*)/
```

Comment:
Ah the Gemfile used to constrain `semantic_puppet` using an array, but that was changed: https://github.com/puppetlabs/puppet/commit/0dd58a942af0e7059605f833191be83a4ad32209#diff-8b7db4d5cc4b8f6dc8feb7030baa2478L31-R31. It might be clearer to change the default constraints being passed, e.g. `location_for(... || "~> 1.0")`. If want to allow arrays to be passed, then maybe check for that case explicitly first, like:

```ruby
if place.is_a?(Array)
  [place, { require: false }]
elsif place =~ /(git...etc/
...
```